crates/gosub_interface/src/draw.rs

```diff
@@ -27,6 +27,19 @@ pub trait TreeDrawer<C: HasDrawComponents> {
     where
         Self: Sized;
 
+    fn from_source(
+        // The initial url that the source was loaded from
+        url: Url,
+        // Actual loaded source HTML
+        source_html: &str,
+        // Layouter that renders the tree
+        layouter: C::Layouter,
+        // Debug flag
+        debug: bool,
+    ) -> impl Future<Output = gosub_shared::types::Result<Self>>
```

Comment:
I don't think this needs to be a future, since the only thing that is async in there is fetching the HTML, which was already done

Reply:
actually, no.. the parsing is also async.. so this needs to be async as well.